Entities/AEmitter.cpp

```diff
@@ -38,13 +38,13 @@ void AEmitter::Clear()
     m_WasEmitting = false;
     m_EmitCount = 0;
     m_EmitCountLimit = 0;
-    m_MinThrottleRange = -1;
-    m_MaxThrottleRange = 1;
+	m_MinThrottleRange = 0.0F;
```

Comment:
Minor note - when using 0 you can just do `m_MinThrottleRange = 0;`
Also, I just noticed this change, I assume defaulting to 0, 0 instead of -1, 1 was deliberate, right? Like you want throttling up/down to not do anything by default for all AEmitters that don't have them explicitly defined?

Reply:
Yeah, drop ship engines (which were the only thing to utilize throttle prior to this PR) have their throttle values defined in vanilla, so it should be apparent that they have to be set in order for throttle to work. I'm certain that every mod that has copied the vanilla drop ship should already have defined these values.

Oh yeah, and so I heard about the zeroes and datatypes. I'll gladly fix that for this PR.

Reply:
Alright, but does it make sense to make throttle not working the default behaviour, even if it's not used? I guess I don't get the logic behind changing this.

Reply:
Yes, it does, like I mentioned - it hasn't really been used anywhere until now. Setting it to zero by default will allow jetpacks (in mods) to function as before (with no throttle alteration) unless explicitly defined.

Reply:
Hmmm, I worry this is kinda screwing over other potential stuff just to make it work well for one use-case. If the goal is to make jetpacks default to 0 +- throttle modifiers, there are other ways to do it while keeping the variables defaulting to useful values for everything else.
But maybe I'm overthinking things and it's fine since, as you say, it's not used anywhere.

Reply:
Yeah I can't believe how much you're overthinking this. Just out of curiosity tho, what "other potential stuff" did you have in mind?

Reply:
Nothing in particular. The problem I have is a general design principle one - how it's done right now obscurely hides the throttle feature by default, instead of making it work by default and adding specific handling to deal with your particular use-case.
This same effect could be achieved for jetpacks by adding some `m_MinThrottleRangeDefined` flag that's set to true in SetMinThrottleRange, then in AHuman and ACrab::SetJetpack, doing `if (!jetpack.m_MinThrottleRangeDefined) { SetMinThrottleRange(0); }` (same for MaxThrottleRange obviously).

It's definitely a bit more work, but I think it's the less side-effecty way to do default behaviour for this stuff, since it doesn't result in turning off the feature by default.

Reply:
I personally don't think this should be a feature that's on by default, but if you wish it to be, the default value should be something milder like `0.5`.

Reply:
Please correct me if I'm wrong, but right now MinThrottleRange affects both how much you can throttle down to (i.e. a value of -1 means you can throttle down to no emissions) and also how much your throttle value matters, so a throttle value of -0.5 and a MinThrottleRange of 0.5 means you'd put out 75% of normal emissions, a throttle value of -0.5 and a MinThrottleRange of 1 means you'd put out 50% of normal emissions, and a throttle value of -1 and a MinThrottleRange of 1 means you'd put out 0% of normal emissions.

Point being, I don't understand the benefit of going with 0.5 over 1. Is there something I'm not understanding or missing here?

Reply:
We can set the default to 1 for both values, but as a result jetpacks (in mods) will change radically with no explanation (or existing .ini reference on the mod's part to explicitly teach how to use the ThrottleRange values).

Jetpacks would default to ranges from 200% to 0%, which effectively screws any jetpack that relies on strong bursts etc.

0.5 as the default value (for both maximum and minimum range) wouldn't be as radical of a change, but I don't really think it makes more sense than just letting jetpacks be as they were without need (for the converter) to add zero values to both properties in order to keep mods functional.

This'll be the full length of me defending mod compatibility lol... I think doubling jetpack power is a hilarious way to break mods, if you insist on it :p

Reply:
Like I said though, with a small amount of work and no real overhead or modder effort we can just keep the full value defaults but make jetpacks that don't set these values in INI/Lua have them set to 0 so there's no change.

I can pop into this branch and do it while we wait for a review from maxim, if you don't object to that. Oh, speaking of maxim, I asked him for an opinion, and he wants to break all the mods just for shits and giggles :D

Reply:
Note - see issue #322